distributed/scheduler.py

```diff
@@ -137,14 +137,21 @@ class ClientState:
 
     """
 
-    __slots__ = ("client_key", "wants_what", "last_seen", "versions")
+    __slots__ = ("client_key", "_hash", "wants_what", "last_seen", "versions")
 
     def __init__(self, client, versions=None):
         self.client_key = client
+        self._hash = hash(client)
         self.wants_what = set()
         self.last_seen = time()
         self.versions = versions or {}
 
+    def __hash__(self):
+        return self._hash
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self.client == other.client
```

Comment:
```suggestion
        return type(self) == type(other) and self.client_key == other.client_key
```

Reply:
I'm somewhat surprised this didn't result in a test failure. I suppose we never do an equality check between `ClientState`s?

Reply:
That doesn't surprise me

Reply:
Fair enough. Thinking about it more, we're using these objects to keep track of state for a given client, not comparing across clients

Reply:
Oops. Thanks James! Added the suggestion.

Yeah this has more to do with meeting the [requirements of a hashable object]( https://docs.python.org/3/glossary.html#term-hashable ).

> An object is hashable if it has a hash value which never changes during its lifetime (it needs a __hash__() method), and can be compared to other objects (it needs an __eq__() method). Hashable objects which compare equal must have the same hash value.